instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -54,7 +54,48 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     super.onStart(attributes, parentContext, request);
-    internalSet(attributes, SemanticAttributes.HTTP_URL, getter.url(request));
+    internalSet(attributes, SemanticAttributes.HTTP_URL, stripSensitiveData(getter.url(request)));
+  }
+
+  @Nullable
+  private static String stripSensitiveData(@Nullable String url) {
+    if (url == null || url.isEmpty()) {
+      return url;
+    }
+
+    int schemeEndIndex = url.indexOf(':');
+
+    if (schemeEndIndex == -1) {
+      // not a valid url
+      return url;
+    }
+
+    int len = url.length();
+    if (len <= schemeEndIndex + 2
+        || url.charAt(schemeEndIndex + 1) != '/'
+        || url.charAt(schemeEndIndex + 2) != '/') {
+      // has no authority component
+      return url;
+    }
+
+    // look for the end of the authority component:
+    //   '/', '?', '#' ==> start of path
+    int index;
+    for (index = schemeEndIndex + 3; index < len; index++) {
+      char c = url.charAt(index);
+      if (c == '/' || c == '?' || c == '#') {
+        break;
+      }
+    }
+
+    int atIndex = url.lastIndexOf("@", index - 1);
+    int questionMarkIndex = url.indexOf("?");
+
+    // '@' char is present and is not a query param
+    if (atIndex == -1 || (questionMarkIndex != -1 && atIndex > index)) {
```

Comment:
I think you should also check whether `@` is the last character and if it is return the url as is. Try with `https://github.com@` java.net.URI parses it so that `github.com@` is authority not userinfo.